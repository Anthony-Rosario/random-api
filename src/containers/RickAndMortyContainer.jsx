import React, { Component } from 'react'
import CharacterList from '../components/app/characters/CharacterList'
import { findCharacters } from '../services/rickandmortyApi'

export default class RickAndMortyContainer extends Component {
  state = {
    characters: [],
    loading: true,
  }

  async componentDidMount() {
    const characters = await findCharacters();
    this.setState({
      characters,
      loading: false,
    });
  }

  render() {
    const { loading, characters } = this.state;
    if(loading) return <h2>Loading...</h2>
    return <CharacterList characters={characters} />;
  }
}
