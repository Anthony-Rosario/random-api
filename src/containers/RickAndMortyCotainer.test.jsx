import React from 'react'
import { render, screen } from '@testing-library/react'
import RickAndMortyContainer from './RickAndMortyContainer'
import { MemoryRouter } from 'react-router-dom'

describe('RickAndMortyContainer', () => {
  it('should render a list of characters', async() => {
    render(<MemoryRouter><RickAndMortyContainer /></MemoryRouter>);
    screen.getByText('Loading...');
    const ul = await screen.findByRole('list', { name: 'characters' });
    expect(ul).toMatchSnapshot();
  });
});