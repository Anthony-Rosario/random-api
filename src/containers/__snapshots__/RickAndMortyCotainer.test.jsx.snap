// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RickAndMortyContainer should render a list of characters 1`] = `
<ul
  aria-label="character list"
>
  <li>
    <a
      href="/1"
    >
      <img
        alt="Rick Sanchez"
        src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
      />
      <p>
        Rick Sanchez
      </p>
      <p>
        Human
      </p>
      <p>
        Alive
      </p>
      <p>
        1
      </p>
    </a>
  </li>
  <li>
    <a
      href="/2"
    >
      <img
        alt="Morty Smith"
        src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
      />
      <p>
        Morty Smith
      </p>
      <p>
        Human
      </p>
      <p>
        Alive
      </p>
      <p>
        2
      </p>
    </a>
  </li>
  <li>
    <a
      href="/3"
    >
      <img
        alt="Summer Smith"
        src="https://rickandmortyapi.com/api/character/avatar/3.jpeg"
      />
      <p>
        Summer Smith
      </p>
      <p>
        Human
      </p>
      <p>
        Alive
      </p>
      <p>
        3
      </p>
    </a>
  </li>
  <li>
    <a
      href="/4"
    >
      <img
        alt="Beth Smith"
        src="https://rickandmortyapi.com/api/character/avatar/4.jpeg"
      />
      <p>
        Beth Smith
      </p>
      <p>
        Human
      </p>
      <p>
        Alive
      </p>
      <p>
        4
      </p>
    </a>
  </li>
  <li>
    <a
      href="/5"
    >
      <img
        alt="Jerry Smith"
        src="https://rickandmortyapi.com/api/character/avatar/5.jpeg"
      />
      <p>
        Jerry Smith
      </p>
      <p>
        Human
      </p>
      <p>
        Alive
      </p>
      <p>
        5
      </p>
    </a>
  </li>
  <li>
    <a
      href="/6"
    >
      <img
        alt="Abadango Cluster Princess"
        src="https://rickandmortyapi.com/api/character/avatar/6.jpeg"
      />
      <p>
        Abadango Cluster Princess
      </p>
      <p>
        Alien
      </p>
      <p>
        Alive
      </p>
      <p>
        6
      </p>
    </a>
  </li>
  <li>
    <a
      href="/7"
    >
      <img
        alt="Abradolf Lincler"
        src="https://rickandmortyapi.com/api/character/avatar/7.jpeg"
      />
      <p>
        Abradolf Lincler
      </p>
      <p>
        Human
      </p>
      <p>
        unknown
      </p>
      <p>
        7
      </p>
    </a>
  </li>
  <li>
    <a
      href="/8"
    >
      <img
        alt="Adjudicator Rick"
        src="https://rickandmortyapi.com/api/character/avatar/8.jpeg"
      />
      <p>
        Adjudicator Rick
      </p>
      <p>
        Human
      </p>
      <p>
        Dead
      </p>
      <p>
        8
      </p>
    </a>
  </li>
  <li>
    <a
      href="/9"
    >
      <img
        alt="Agency Director"
        src="https://rickandmortyapi.com/api/character/avatar/9.jpeg"
      />
      <p>
        Agency Director
      </p>
      <p>
        Human
      </p>
      <p>
        Dead
      </p>
      <p>
        9
      </p>
    </a>
  </li>
  <li>
    <a
      href="/10"
    >
      <img
        alt="Alan Rails"
        src="https://rickandmortyapi.com/api/character/avatar/10.jpeg"
      />
      <p>
        Alan Rails
      </p>
      <p>
        Human
      </p>
      <p>
        Dead
      </p>
      <p>
        10
      </p>
    </a>
  </li>
  <li>
    <a
      href="/11"
    >
      <img
        alt="Albert Einstein"
        src="https://rickandmortyapi.com/api/character/avatar/11.jpeg"
      />
      <p>
        Albert Einstein
      </p>
      <p>
        Human
      </p>
      <p>
        Dead
      </p>
      <p>
        11
      </p>
    </a>
  </li>
  <li>
    <a
      href="/12"
    >
      <img
        alt="Alexander"
        src="https://rickandmortyapi.com/api/character/avatar/12.jpeg"
      />
      <p>
        Alexander
      </p>
      <p>
        Human
      </p>
      <p>
        Dead
      </p>
      <p>
        12
      </p>
    </a>
  </li>
  <li>
    <a
      href="/13"
    >
      <img
        alt="Alien Googah"
        src="https://rickandmortyapi.com/api/character/avatar/13.jpeg"
      />
      <p>
        Alien Googah
      </p>
      <p>
        Alien
      </p>
      <p>
        unknown
      </p>
      <p>
        13
      </p>
    </a>
  </li>
  <li>
    <a
      href="/14"
    >
      <img
        alt="Alien Morty"
        src="https://rickandmortyapi.com/api/character/avatar/14.jpeg"
      />
      <p>
        Alien Morty
      </p>
      <p>
        Alien
      </p>
      <p>
        unknown
      </p>
      <p>
        14
      </p>
    </a>
  </li>
  <li>
    <a
      href="/15"
    >
      <img
        alt="Alien Rick"
        src="https://rickandmortyapi.com/api/character/avatar/15.jpeg"
      />
      <p>
        Alien Rick
      </p>
      <p>
        Alien
      </p>
      <p>
        unknown
      </p>
      <p>
        15
      </p>
    </a>
  </li>
  <li>
    <a
      href="/16"
    >
      <img
        alt="Amish Cyborg"
        src="https://rickandmortyapi.com/api/character/avatar/16.jpeg"
      />
      <p>
        Amish Cyborg
      </p>
      <p>
        Alien
      </p>
      <p>
        Dead
      </p>
      <p>
        16
      </p>
    </a>
  </li>
  <li>
    <a
      href="/17"
    >
      <img
        alt="Annie"
        src="https://rickandmortyapi.com/api/character/avatar/17.jpeg"
      />
      <p>
        Annie
      </p>
      <p>
        Human
      </p>
      <p>
        Alive
      </p>
      <p>
        17
      </p>
    </a>
  </li>
  <li>
    <a
      href="/18"
    >
      <img
        alt="Antenna Morty"
        src="https://rickandmortyapi.com/api/character/avatar/18.jpeg"
      />
      <p>
        Antenna Morty
      </p>
      <p>
        Human
      </p>
      <p>
        Alive
      </p>
      <p>
        18
      </p>
    </a>
  </li>
  <li>
    <a
      href="/19"
    >
      <img
        alt="Antenna Rick"
        src="https://rickandmortyapi.com/api/character/avatar/19.jpeg"
      />
      <p>
        Antenna Rick
      </p>
      <p>
        Human
      </p>
      <p>
        unknown
      </p>
      <p>
        19
      </p>
    </a>
  </li>
  <li>
    <a
      href="/20"
    >
      <img
        alt="Ants in my Eyes Johnson"
        src="https://rickandmortyapi.com/api/character/avatar/20.jpeg"
      />
      <p>
        Ants in my Eyes Johnson
      </p>
      <p>
        Human
      </p>
      <p>
        unknown
      </p>
      <p>
        20
      </p>
    </a>
  </li>
</ul>
`;
